filetests/isa/x86/legalize-custom.clif

```diff
@@ -90,3 +90,56 @@ block0(v0: f32, v1: f32):
     ; check: $done(v2: f32):
     ; nextln: return v2
 }
+
+function %ineg_legalized_i8() {
+block0:
+    v0 = iconst.i8 1
+    v1 = ineg v0
+    ; check: v2 = iconst.i32 1
+    ; nextln: v0 = ireduce.i8 v2
+    ; nextln: v3 = iconst.i8 0
+    ; nextln: v4 = uextend.i32 v3
+    ; nextln: v5 = uextend.i32 v0
+    ; nextln: v6 = isub v4, v5
+    ; nextln: v1 = ireduce.i8 v6
+
+    return
+}
+
+function %ineg_legalized_i16() {
+block0:
+    v0 = iconst.i16 1
+    v1 = ineg v0
+    ; check: v2 = iconst.i32 1
+    ; nextln: v0 = ireduce.i16 v2
+    ; nextln: v3 = iconst.i16 0
+    ; nextln: v4 = uextend.i32 v3
+    ; nextln: v5 = uextend.i32 v0
+    ; nextln: v6 = isub v4, v5
+    ; nextln: v1 = ireduce.i16 v6
+
+    return
+}
+
+function %ineg_legalized_i32() {
+block0:
+    v0 = iconst.i32 1
+    v1 = ineg v0
+    ; check: v0 = iconst.i32 1
+    ; nextln: v2 = iconst.i32 0
+    ; nextln: v1 = isub v2, v0
+
+    return
+}
+
+function %ineg_legalized_i64() {
+block0:
+    v0 = iconst.i64 1
+    v1 = ineg v0
+    ; check: v2 = iconst.i32 1
+    ; check: v3 = iconst.i32 0
+    ; check: v0 = iconcat v2, v3
```

Comment:
@bjorn3, I would not expect this `iconcat`; can't we directly encode an `iconst.i64`?

Reply:
This file has `target i686` before `target x86_64`. How is that handled? Does it test both or only the first?

Reply:
Learn something new every day: apparently parsing tests [accumulates the targets](https://github.com/bytecodealliance/cranelift/blob/master/cranelift-reader/src/parser.rs#L992-L1025) and `test_tuples` will expand the different targets (in this case since the legalizer `needs_isa` we should have a test tuple for each `target` from [here](https://github.com/bytecodealliance/cranelift/blob/master/cranelift-filetests/src/runone.rs#L121-L125)). 

But then this doesn't make sense: how could these checks pass when `target x86_64` is set? Is that the build failure? No, it looks like the current build failure happens because `ineg` isn't getting legalized at all for `i64`?

Reply:
Yeah the build failure is happening because `ineg` still isn't getting legalized for `i64`, I still haven't figured out why that is. This must be because I need to extend another transform group right? That was the reason for `i8` and `i16` not getting legalized initially (I think)

Reply:
There is also the `narrow` group for ints > pointer size.